expression/builtin_cast_vec.go

```diff
@@ -875,11 +875,42 @@ func (b *builtinCastJSONAsStringSig) vecEvalString(input *chunk.Chunk, result *c
 }
 
 func (b *builtinCastDurationAsRealSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeFloat64(n, false)
+	result.MergeNulls(buf)
+	f64s := result.Float64s()
+
+	var duration types.Duration
+	fsp := int8(b.tp.Decimal)
```

Comment:
Should it be `b.args[0].GetType().Decimal`?

Reply:
Oh...I'm sorry...

Comment addressed, PTAL, Thanks. @qw4990